src/Scheduler/ImmediateScheduler.php

```diff
@@ -48,20 +49,18 @@ public function scheduleRecursive(callable $action)
 
     /**
      * @inheritDoc
+     * @throws \Exception
      */
-    public function schedulePeriodic(callable $action, $delay, $period)
+    public function schedulePeriodic(callable $action, $delay, $period): DisposableInterface
     {
-        throw new \Exception("ImmediateScheduler does not support a non-zero delay.");
+        throw new \Exception('ImmediateScheduler does not support a non-zero delay.');
```

Comment:
On second thought - I don't think there is much to be gained by adding the specific exception here.